### ISSUE TITLE
Cannot use underscor (_) in database name

### ISSUE THREAD

**Issue by arnemorken** [agershun/alasql#759]
Consider:
  alasql("CREATE localStorage DATABASE IF NOT EXISTS "+dbName);
  alasql("ATTACH localStorage DATABASE "+dbName);

If dbName contains an underscore (_), the second statement fails with this message in console:
  TypeError: LS.get(...) is null
  db.tables = LS.get(lsdbid).tables;

**Comment by mathiasrw** [comment]
Thank you for a very well described bug!

**Comment by arnemorken** [comment]
Seems the problem is not related to underscore, but it occurs when the lines in webappstore.sqlite containing my database is removed. But the real problem seems to be that LS.get(lsdbid) returns null, and so LS.get(lsdbid).tables fails. A check for null (or a try/catch) should fixc the probem, I guess?

**Comment by mathiasrw** [comment]
Hmm. So you remove the file while its attatched?

**Comment by arnemorken** [comment]
Not exactly. Im creating a backup/restore function for my program. I removed the lines representing my database from the webappstore2 table (using the SQLiteManager plugin for Firefox) in order to see how my program would react if no database was found. That's when I get the error message I reported. 

This might be an obscure bug, and I can easily get around it by using a try/catch. Anyway, the offending line 16714 should be fixed, I think.

**Comment by mathiasrw** [comment]
Trying to understand here: `LS.get(lsdbid)` retuning null beacuse you removed details from the .sqlite file?

To use SQLite files you need to use `ATTACH SQLITE DATABASE` - but your example is having `ATTACH localStorage DATABASE`

Have a look at https://github.com/agershun/alasql/wiki/SQLite and let me know if this helps you.

**Comment by arnemorken** [comment]
>retuning null beacuse you removed details from the .sqlite file?
So it would seem.

>To use SQLite files you need to use ATTACH SQLITE DATABASE - but your example is having ATTACH localStorage DATABASE
With my code (ATTACH localStorage DATABASE), the data actually goes into the webappstore2 table in Firefox's webappstore.sqlite file...

**Comment by mathiasrw** [comment]
> With my code (ATTACH localStorage DATABASE), the data actually goes into the webappstore2 table in Firefox's webappstore.sqlite file...

Ahhhh, I get what you are saying

**Comment by arnemorken** [comment]
I changed the name of the issue to reflect the real problem.
BTW, thanks for a great plugin! :)  :+1:
